src/types.ts

```diff
@@ -0,0 +1,199 @@
+export type Unparsed = unknown
+
+export type Uint = (number | bigint) & { __type: 'uint' }
+export type Int = (number | bigint) & { __type: 'int' }
+
+export type MaxlenString<N> = string & { __maxLength: N }
+export type FixlenBuffer<N> = Buffer & { __length: N }
+export type MaxlenBuffer<N> = Buffer & { __maxLength: N }
+
+export const ADDRESS_KEY_HASH_LENGTH = 28
+export const SCRIPT_HASH_LENGTH = 28
+export const GENESIS_DELEGATE_HASH_LENGTH = 28
+export const POOL_KEY_HASH_LENGTH = 28
+export const GENESIS_HASH_LENGTH = 28
+export const VRF_KEY_HASH_LENGTH = 32
+export const METADATA_HASH_LENGTH = 32
+export const TX_ID_HASH_LENGTH = 32
+
+export const IPV4_LENGTH = 4
+export const IPV6_LENGTH = 16
+
+export const ASSET_NAME_MAX_LENGTH = 32
+export const URL_MAX_LENGTH = 64
+export const DNS_NAME_MAX_LENGTH = 64
+
+export type Address = Buffer
+export type RewardAccount = Buffer
```

Comment:
I would stick to the CDDL in this case as we are kind of also validating against that. And if I am not mistaken this can actually be a `FixlenBuffer<29>`.

Reply:
You are right I didn't realize it from the CDDL at first because it's in the "commented" out part:
```
; address format:
; [ 8 bit header | payload ];
...
; 1110: reward account: keyhash28
; 1111: reward account: scripthash28
```
So yes, reward account should be a `FixlenBuffer<29>`